File: src/services/inventoryService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/inventory';

const getAllInventories = () => {
    return axios.get(API_URL);
};

const updateInventory = (inventory) => {
    return axios.put(API_URL, inventory);
};

export default {
    getAllInventories,
    updateInventory
};

File: inventory-2/src/main/java/com/example/demo/service/InventoryService.java
package com.example.demo.service;

import com.example.demo.entity.Inventory;
import com.example.demo.repository.InventoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class InventoryService {

    @Autowired
    private InventoryRepository InventoryRepository;

    public List<Inventory> getAllInventories() {
        return InventoryRepository.findAll();
    }

    public Inventory updateInventory(Inventory inventory) {
        return InventoryRepository.save(inventory);
    }
    
    public Inventory saveInventory(Inventory inventory) {
        return InventoryRepository.save(inventory);
    }
    public Inventory createInventory(Inventory inventory) {
        return InventoryRepository.save(inventory);
    }
}
